### PR TITLE
SDI-301 A few more receive scenarios to show pattern

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt [hunk 1]
```diff
@@ -20,25 +24,76 @@ class PrisonerMovementsEventService(private val domainEventEmitter: HmppsDomainE
...
+      )
     }
   }
 
   private fun calculateMovementChange(previousPrisonerSnapshot: Prisoner?, prisoner: Prisoner): PossibleMovementChange {
     return previousPrisonerSnapshot.let {
-      // TODO: when we do a few more of these we can extract a nice state machine pattern
-      if (previousPrisonerSnapshot?.inOutStatus == "TRN" && prisoner.inOutStatus == "IN") {
-        TransferIn(prisoner.prisonerNumber!!, prisoner.prisonId!!)
+      val prisonerNumber = prisoner.prisonerNumber!!
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:44]: any suggestions for alternative for `if else` is more than welcome

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt [hunk 2]
```diff
@@ -20,25 +24,76 @@ class PrisonerMovementsEventService(private val domainEventEmitter: HmppsDomainE
...
+      is CourtReturn -> domainEventEmitter.emitPrisonerReceiveEvent(
+        offenderNo = movementChange.offenderNo,
+        reason = movementChange.reason,
+        prisonId = movementChange.prisonId,
+      )
+
+      is TAPReturn -> domainEventEmitter.emitPrisonerReceiveEvent(
+        offenderNo = movementChange.offenderNo,
+        reason = movementChange.reason,
+        prisonId = movementChange.prisonId,
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:37]: Won't these events be identical?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt (hunk 1 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:44]: You could use `when` - fewer lines but each harder to read. Not much in it.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:37]: Oh yeah, that was a stupid comment. Pleae ignore.

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**